src/neuroconv/nwbconverter.py

```diff
@@ -60,15 +60,27 @@ def validate_source(cls, source_data: Dict[str, dict], verbose: bool = True):
         """Validate source_data against Converter source_schema."""
         cls._validate_source_data(source_data=source_data, verbose=verbose)
 
-    def __init__(self, source_data: Dict[str, dict], verbose: bool = True):
+    def __init__(self, source_data: Dict[str, dict] = None, data_interfaces: List = None, verbose: bool = True):
         """Validate source_data against source_schema and initialize all data interfaces."""
         self.verbose = verbose
-        self._validate_source_data(source_data=source_data, verbose=self.verbose)
-        self.data_interface_objects = {
-            name: data_interface(**source_data[name])
-            for name, data_interface in self.data_interface_classes.items()
-            if name in source_data
-        }
+
+        if source_data is not None:
+            self._validate_source_data(source_data=source_data, verbose=self.verbose)
+
+            self.data_interface_objects = {
+                name: data_interface(**source_data[name])
+                for name, data_interface in self.data_interface_classes.items()
+                if name in source_data
+            }
+
+        elif data_interfaces is not None:
+
+            self.data_interface_objects = {
+                f"{interface.__class__.__name__}{index}": interface for index, interface in enumerate(data_interfaces)
+            }
```

Comment:
I see you are using `index` here to ensure all of the DataInterfaces have unique names. I think it might be better to start without a number and then add numbers if you have more than one of the same type. I think that would be a bit too complicated for a dictionary comprehension though

Reply:
I also think it would be nice to allow the user to input data_interfaces as a dictionary

Reply:
I agree with this. My initial instinct was to have the interface variable name as the name. That is, if someone defined:

```
interface_phy = PhySortingInterface(folder_path=folder_path, verbose=False)
interface_spikeglx = SpikeGLXRecordingInterface(file_path=file_path, verbose=False)
```
And passed those to the NWBConverter as a list:
```
converter = NWBConverter(data_interfaces=[interface_spikeglx, interface_phy], verbose=False)
```

Then their names in the `self.data_interface_objects` should be `interface_phy` and `interface_spikeglx`. This will care of uniqueness of name for free and is also a natural reference that the user already knows because they defined themselves.  The problem with this is that it can only be done very easily after python=3.8, python 3.7 requires a hacky solution.

Reply:
> I also think it would be nice to allow the user to input data_interfaces as a dictionary

Yep, this make sense. We can have a general variable called `data_interfaces` and users can pass either a dict or a list. If there is a list then we ourselves determine our names with the logic mentioned above.

Reply:
I implemented both of these things in the last commit.